autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/SectionSelect/SectionSelect.tsx

```diff
@@ -1,36 +1,110 @@
+  SectionSelected, SortType, SortTypeLabels, DefaultSortTypeDirections, SectionFilter,
 } from '../../../../../../types/CourseCardOptions';
 import { RootState } from '../../../../../../redux/reducer';
 import * as styles from './SectionSelect.css';
 import ProfessorGroup from './ProfessorGroup';
 import SmallFastProgress from '../../../../../SmallFastProgress';
+import BasicOptionRow from './BasicOptionRow';
+import BasicCheckbox from './BasicCheckbox';
 
 interface SectionSelectProps {
   id: number;
 }
 
+const ExpansionPanel = withStyles({
+  root: {
+    boxShadow: 'none',
+    display: 'flex',
+    flexDirection: 'column',
+    height: '100%',
+    width: '100%',
+  },
+  expanded: {},
+})(ExpansionPanelBase);
+
+const ExpansionPanelSummary = withStyles({
+  root: {
+    marginBottom: -1,
+    minHeight: 36,
+    '&$expanded': {
+      minHeight: 36,
+    },
+  },
+  content: {
+    margin: '0',
+    '&$expanded': {
+      margin: '0',
+    },
+    overflow: 'hidden',
+    // bar between arrow and title
+    '&::after': {
+      content: '""',
+      height: 2,
+      borderWidth: '6px 0 0 0',
+      borderColor: 'transparent',
+      borderStyle: 'solid',
+      boxShadow: 'inset 0 0 10px 10px #919191',
+      margin: 'auto -14px auto 12px',
+      flexGrow: 1,
+    },
+  },
+  expandIcon: {
+    // the typography has a slight top margin
+    marginTop: 6,
+    padding: '4px 16px',
+  },
+  expanded: {},
+})(ExpansionPanelSummaryBase);
+
+const ExpansionPanelDetails = withStyles({
+  root: {
+    flexGrow: 1,
+  },
+})(ExpansionPanelDetailsBase);
+
 const SectionSelect: React.FC<SectionSelectProps> = ({ id }): JSX.Element => {
-  const sections = useSelector<RootState, SectionSelected[]>(
-    (state) => state.termData.courseCards[id].sections,
-  );
-  // to show loading symbol when needed
-  const reduxSortType = useSelector<RootState, SortType>(
-    (state) => state.termData.courseCards[id].sortType,
-  );
-  const reduxSortIsDescending = useSelector<RootState, boolean>(
-    (state) => state.termData.courseCards[id].sortIsDescending,
-  );
+  const {
+    course = '',
+    sections,
+    // to show loading symbol when needed
+    sortType: reduxSortType,
+    sortIsDescending: reduxSortIsDescending,
+    // filters
+    hasHonors = false,
+    hasRemote = false,
+    hasAsynchronous = false,
+    honors = SectionFilter.NO_PREFERENCE,
+    remote = SectionFilter.NO_PREFERENCE,
+    asynchronous = SectionFilter.NO_PREFERENCE,
+    includeFull = false,
+  }: {
+    course: string;
+    sections: SectionSelected[];
+    // to show loading symbol when needed
+    sortType: SortType;
+    sortIsDescending: boolean;
+    // filters
+    hasHonors: boolean;
+    hasRemote: boolean;
+    hasAsynchronous: boolean;
+    honors: SectionFilter;
+    remote: SectionFilter;
+    asynchronous: SectionFilter;
+    includeFull: boolean;
+  } = useSelector<RootState, any>((state) => state.termData.courseCards[id]);
```

Comment:
This is a good example of when *not* to use destructuring: these declarations take 30 lines and only serve to make the code more ambiguous imo (since it's not immediately clear all these variables are all retrieved from course card data)

Reply:
Do you want me to go back to how it was before with the multiple `useSelector()`'s?

Reply:
I would recommend using a single `useSelector`, saving that as a temporary variable called something like `courseCardData`, and then pulling the fields individually from the `courseCardData` object